token_dispenser_client/token_dispenser_client.py

```diff
@@ -0,0 +1,216 @@
+        raise e
+
+
+def invoke_lambda(input_params_json, lambda_arn):
+    """
+    Invokes an AWS Lambda function with input parameters from a JSON string.
+
+    Args:
+        input_params_json (str): A JSON string containing input parameters.
+        lambda_arn (str): The ARN of the Lambda function to invoke.
+
+    Returns:
+        str: The JSON-formatted response from the Lambda function, or an error message.
+    """
+    if not lambda_arn:
+        return "Error: lambda_arn not provided"
+
+    try:
+        input_params = json.loads(input_params_json)
+    except json.JSONDecodeError:
+        return f'Error: Invalid JSON : {input_params_json}'
+
+    try:
+        response = lambda_client.invoke(
+            FunctionName=lambda_arn,
+            InvocationType='RequestResponse',  # Synchronous invocation
+            Payload=json.dumps(input_params).encode('utf-8')
+        )
+
+        payload = response['Payload'].read().decode('utf-8')
+
+        # Check for function error
+        if 'FunctionError' in response:
+            return f"Error: Lambda function error: {payload}"
+
+        return payload
+
+    except Exception as e:
+        return f"Error: Lambda invocation failed: {e}"
+
+def validate_input(client_id:str, minimum_alive_secs:int) -> str:
+    err_msg=''
+    if client_id is None or client_id.strip() == '':
+        err_msg='Error: client_id is required'
+
+    pattern = re.compile(r'^[a-zA-Z0-9]{3,32}$')
+    if client_id and not pattern.match(client_id):
+        err_msg='client_id must be between length 3-32 with pattern [a-zA-Z0-9]{3,32}'
+
+    if minimum_alive_secs is not None and (not isinstance(minimum_alive_secs, int)):
+        err_msg = f'{err_msg}\n Minimum alive interval must be an integer'
+
+    if isinstance(minimum_alive_secs, int) and (minimum_alive_secs > 3300 or minimum_alive_secs < 0):
+        err_msg = f'{err_msg}\n Minimum alive interval must be an integer between 1 and 3300'
+    return err_msg
+
+
+def get_tds_arn(ssm_name:str) -> str:
+    """
+    :param ssm_name:
+    :return:
+    err_msg:str : error message if there is error
+    arn:str     : ARN of TDS lambda
+    """
+    err_msg=''
+    if ssm_name is None:
+        ssm_name=_DEFAULT_SSM_PATH_
+        tds_arn = ''
+        params = get_parameters_by_path(path=ssm_name, with_decryption=False)
+        size = params.__len__()
+        if size < 1:
+            raise ValueError(f"Not able to find tds arn for: {ssm_name}")
+        elif size > 1:
+            raise ValueError(f"Found more than one tds arn for: {ssm_name}")
+        elif size == 1:
+            items_list = list(params.items())
+            first_item = items_list[0]
+            first_key, tds_arn = first_item
+        return tds_arn
+    else:
+        # if user provides a name, the code trusts it as a full name and let system fail if
+        # provided name is not correct
+        return get_parameter_by_name(name=ssm_name, with_decryption=False)
```

Comment:
We don't need the `else` since the `if` statement above returns and will exit the function after its execution